backend/app/utils.py

```diff
@@ -96,6 +97,21 @@ def recurse(t, parent_key=""):
     recurse(d)
     return d
 
+def remove_masked_key_value(d):
```

Comment:
I was wondering if we could do better at `EncryptedStr` level. The idea would be this: at instantiation, if the value passed to the constructor corresponds to the `SECRET_MASK`, return `None`. This way, this could be handled by the `.dict` method with the `exclude_none=True` parameter.

```py
class EncryptedStr(str):
    """Representation of an encrypted string."""

    def __new__(cls, value):
        """
        Create a string object from the given value.

        We try to decrypt it to check if it's already an encrypted value,
        e.g. if it comes from the database. If it is, we just set it as it.

        Otherwise, we encrypt it immediately.
        """
        if value == c.SECRET_MASK:
            return None
        try:
            security.decrypt_password(value)
            return super(EncryptedStr, cls).__new__(cls, value)
        except InvalidToken:
            return super(EncryptedStr, cls).__new__(cls, security.encrypt_password(value))
```

I've not tested it, but could worth a try.

Reply:
Works like a charm 💪